.github/workflows/ci.yml

```diff
@@ -1,7 +1,13 @@
 name: doctest-CI
+
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
```

Comment:
I don't know whether restricting `pull_request` to `main` is necessary. But then, PRs are only made to `main` on this repo, so it looks good!